### PR TITLE
The (very) Big Docs Sync !

### PR DESCRIPTION
**Pull request by silopolis**
Full pass on english source docs.
Full French translation sync of existing _fr.adoc files.
Final Spanish translation sync of existing _es.adoc files.

### CODE REVIEW

**Comment by silopolis** [comment]
Hi guys,

It's finally there, and HUGE :sweat_smile: 
PDFs build locally, built still have an issue with HTML docs breaking on a couple of image tags, again :angry: 
Pushed it to see what GH says...

**Comment by hansu** [comment]
Does this include #1719 and #1720?

**Comment by silopolis** [comment]
Le sam. 30 avr. 2022 à 23:39, Hans ***@***.***> a écrit :

> Does this include #1719 <https://github.com/LinuxCNC/linuxcnc/pull/1719>
> and #1720 <https://github.com/LinuxCNC/linuxcnc/pull/1720>?
>

Yes, and much more!

**Comment by hansu** [comment]
:+1: 
So they can be closed?

**Comment by silopolis** [comment]
Le dim. 1 mai 2022 à 10:21, Hans ***@***.***> a écrit :

> 👍
> So they can be closed?
>
Yes ! I'll delete these branches once all this will be merged anyway as all
has been rebased/merged/reordered/squashed into here.

**Comment by silopolis** [comment]
Aaaaand it builds! 🎉

**Comment by andypugh** [comment]
Great. Do you need anything from me on this? (For example, editing files on www.linuxncnc.org to point to the right things)

**Comment by silopolis** [comment]
Hi Andy,

Hope you had a nice we

Le lun. 2 mai 2022 à 02:02, andypugh ***@***.***> a écrit :

> Great. Do you need anything from me on this? (For example, editing files
> on www.linuxncnc.org to point to the right things)
>
I can't tell myself sadly but may be the case as translated docs are
generated in subdirectories.

I have to study the build system for docs to better understand it...

I also don't know how this integrates with wlo. Can't you point me
somewhere to better understand this too, please?

Have a very nice week team 🙂

**Comment by andypugh** [comment]
I think that the buildbot dumps the output into the docs directory on wlo. The link on the homepage takes you to the page that is generated by this code in the LinuxCNC main repository:
https://github.com/LinuxCNC/linuxcnc/blob/master/docs/src/index.tmpl
Which has versions in other languages, that I assume were hand-built.
I don't know if there is a way to include that html source in weblate? 
At the moment the [README.adoc](https://github.com/LinuxCNC/linuxcnc/blob/master/docs/README.adoc) says that index.tmpl must be created to add a new document set. 
I don't think that the Submakefile does much to the .tmpl other than append a footer and update the version. 
index.tmpl is treated a little differently to the index_lang.tmpl files. I rather feel that renaming to index_en.tmpl might be neater.

**Comment by smoe** [comment]
> I think that the buildbot dumps the output into the docs directory on wlo.

I also understand that http://linuxcnc.org/docs/devel/html/ points to the latest builds from master. When someone performs a functional change to the source tree (in src/) and changes the documentation in the same PR (in docs/) then the PR would be against master. How do we inform weblate about that change? By merging the change with docs-devel?

Once I think to have understood how HAL works on the RPi and have some funny ladder "scripts" to control a few LEDs and switches for a small tutorial, then I tend to think that I should prepare a PR against docs-devel, since this is not synced with any development on master. Am I right? Or am I complicating things? 

Last thought - once the .po system works - do we truly need the docs-devel branch, since weblate works with PRs that are build-tested anyway?

> The link on the homepage takes you to the page that is generated by this code in the LinuxCNC main repository: https://github.com/LinuxCNC/linuxcnc/blob/master/docs/src/index.tmpl Which has versions in other languages, that I assume were hand-built. I don't know if there is a way to include that html source in weblate? At the moment the [README.adoc](https://github.com/LinuxCNC/linuxcnc/blob/master/docs/README.adoc) says that index.tmpl must be created to add a new document set. I don't think that the Submakefile does much to the .tmpl other than append a footer and update the version. index.tmpl is treated a little differently to the index_lang.tmpl files. I rather feel that renaming to index_en.tmpl might be neater.

I agree that _en would be nice. And we also need something for the weblate community to see the effects of their changes.

**Comment by hansu** [comment]
> I also understand that http://linuxcnc.org/docs/devel/html/ points to the latest builds from master. When someone performs a functional change to the source tree (in src/) and changes the documentation in the same PR (in docs/) then the PR would be against master. How do we inform weblate about that change? By merging the change with docs-devel?

Maybe we are waiting until docs-devel is in master to set up Weblate.  Then after every commit on master, master is merged to Weblate (like it is working actual with Weblate). Or should we do a test round on docs-devel with Weblate?

> Once I think to have understood how HAL works on the RPi and have some funny ladder "scripts" to control a few LEDs and switches for a small tutorial, then I tend to think that I should prepare a PR against docs-devel, since this is not synced with any development on master. Am I right? Or am I complicating things?

I would suggest keeping it in a separate branch until docs-devel is merged into master and then to a PR to master.


> Last thought - once the .po system works - do we truly need the docs-devel branch, since weblate works with PRs that are build-tested anyway?

If Jérémies branch is merged and everything works as it should, docs-devel can be merged into master.

**Comment by silopolis** [comment]
Hi guys !

So if yall validate this tiny little PR and agree, merging it should check tasks in #1713 up to 5th included :)

@hansu up to you Master ;-)

**Comment by silopolis** [comment]
Le mar. 3 mai 2022 à 11:21, Hans ***@***.***> a écrit :

> Alright, thanks! But unfortunately I will not have time to look into this
> before tomorrow evening.
>
Can I help and prepare things for you ?
I tried but I'm making git knots in my head and can't see how I could make
things so that you don't have to resolve conflicts too :/

Would it help if I rebase docs-devel on master locally, and then rebase
this branch in turn on docs-devel ?
You would still get conflicts but maybe you could accept all incoming
changes "blindly" ?

If you guys have any guidance to share, I'll gladly give it another shot...

**Comment by hansu** [comment]
>Can I help and prepare things for you ?

No, thanks, I only have to go through the changes.

>You would still get conflicts but maybe you could accept all incoming
changes "blindly" ?

No merge conflicts here...

**Comment by silopolis** [comment]
Le mar. 3 mai 2022 à 20:23, Hans ***@***.***> a écrit :

> Can I help and prepare things for you ?
>
> No, thanks, I only have to go through the changes.
>

OK, sending you a virtual six pack to bear the burden! 😊

You would still get conflicts but maybe you could accept all incoming
> changes "blindly" ?
>
> No merge conflicts here...
>

Rebasing docs-devel onto master?, 🤔

File: docs/src/config/stepper-quickstart.adoc [hunk 1]
```diff
@@ -140,15 +141,15 @@ latexmath:[
...
+----
 ............................................
     Stepper         = 200 steps per revolution
     Drive           =   8 micro steps per step
     Motor Teeth     =  30
     Leadscrew Teeth =  90
     Leadscrew Pitch =   5.00 mm per turn
 ............................................
-====
+----
```
Comment (hansu) [inline docs/src/config/stepper-quickstart.adoc:153]: Sometimes it feels like ping pong.
For this example: Steffen misshaped the formatting of this block, then I corrected it, and then you are coming again with this :see_no_evil: :smile:

File: docs/src/drivers/mb2hal.adoc [hunk 2]
```diff
@@ -103,56 +139,82 @@ Modbus transaction function code (see link:https://modbus.org/specs.php[specific
...
 
+FIXME Files missing
+
 Click link:mb2hal_HOWTO.ini[here] to download.
 
 [source,{ini}]
 ----
 include::mb2hal_HOWTO.ini[]
 ----
+////
```
Comment (hansu) [inline docs/src/drivers/mb2hal.adoc:217]: The file is copied by the Submakefile, so no problem here...

File: docs/src/gcode/tool-compensation.adoc [hunk 3]
```diff
@@ -248,13 +248,13 @@ The io program provides hal output pins to facilitate toolchanger management:
...
 .Manual Tool Changer
-Manual tool changer (you change the tool by hand) is treated like a fixed location tool changer.
-Using the manual tool changer only makes sense if you have toolholders that remain with the tool (Cat, NMTB, Kwik Switch, etc.) so that the tool position is preserved in the spindle.
-Machines with R8 or router collet type toolholders do not retain the tool location and the manual tool changer should not be used.
-Manual toolchanges can be aided by a hal configuration that employs the userspace program *hal_manualtoolchange* and is typically specified in an ini file
+Manual tool changer (you change the tool by hand) is treated like a
+fixed location tool changer.  Manual toolchanges can be aided by
+a hal configuration that employs the userspace program
+*hal_manualtoolchange* and is typically specified in an ini file
 with ini statements:
```
Comment (hansu) [inline docs/src/gcode/tool-compensation.adoc:258]: Why did you remove/shorten this?

File: docs/src/config/stepper-quickstart.adoc (hunk 1 above)
Comment (silopolis) [reply docs/src/config/stepper-quickstart.adoc:153]: Worst is that I don't understand why I reverted it :facepalm:

FIXED

OT: commenting straight from Codium... Men I like that IDE, not perfect for sure, but so many cool things :)

File: docs/src/drivers/mb2hal.adoc (hunk 2 above)
Comment (silopolis) [reply docs/src/drivers/mb2hal.adoc:217]: FIXED thanks

File: docs/src/config/stepper-quickstart.adoc (hunk 1 above)
Comment (hansu) [reply docs/src/config/stepper-quickstart.adoc:153]: >Worst is that I don't understand why I reverted it 

Because it is not obvious that I changed it. And it was still existing in the other format in `_es`. And my change was more intended as a comment to propagate it to ES as well, but I know it is hard to keep the overview on that mass of files and changes ...

You don't have to fix anything here, I do that. It's just for information/questioning.

File: docs/src/gui/gmoccapy.adoc [hunk 4]
```diff
@@ -1432,17 +1449,18 @@ in absolute coordinates
...
-* 4º eje DRO
-* 4º eje boton Jog
-* 4º eje botón de referencia (home)
-* Columna del 4º eje en la página offset.
-* Columna del 4º eje en el editor de herramientas.
+* 4th axis DRO
+* 4th axis jog button
+* 4th axis homing button
+* 4th axis column on the offset page.
+* 4th axis column in the tool editor.
```
Comment (hansu) [inline docs/src/gui/gmoccapy.adoc:1464]: Good catch! I completely missed this.

File: docs/src/gcode/tool-compensation.adoc (hunk 3 above)
Comment (silopolis) [reply docs/src/gcode/tool-compensation.adoc:258]: I have absolutly no remembering and can not imagine myself do it !???

FIXED thanks

**Comment by hansu** [comment]
Didn't you miss my comment?

>You don't have to fix anything here, I do that. It's just for information/questioning. (https://github.com/LinuxCNC/linuxcnc/pull/1724#discussion_r865136283)

**Comment by silopolis** [comment]
> Didn't you miss my comment?

Seen it after fixing locally. Having the reviews integrated into editor makes it so easier :)

File: docs/src/gui/gstat.adoc [hunk 5]
```diff
@@ -170,429 +177,550 @@ class HandlerClass:
...
+  Returns nothing +
+  Sent every 100 ms.
+
+*state-estop*::
+  Returns nothing +
+  Sent when linuxcnc is goes into estop.
+
+*state-estop-reset*::
+  Returns nothing +
+  Sent when linuxcnc comes out of estop.
```
Comment (hansu) [inline docs/src/gui/gstat.adoc:193]: You have done a lot of great improvements, but on these I don't agree. IMHO the original formatting was more clearly arranged.  i have reverted the corresponding lines on my branch.

Comment (silopolis) [reply docs/src/gui/gstat.adoc:193]: This is only a first pass actually.
Final idea is more like

``̀
*`hard-limits-tripped`*::
  * _Returns_: `bool, Python List` +
  * _Sent_: When any hard limit is tripped. +
  * _Comment_:
    * `bool` indicates if any limit is tripped,
    * `list` shows all available joint's current limit values.
```

Comment (hansu) [reply docs/src/gui/gstat.adoc:193]: But the compact notation has also advantages...

This could be a compromise:

- state-estop 
_returns (nothing)_ - Sent when linuxcnc is goes into estop.

- hard-limits-tripped  
      _returns (bool, Python List)_ -  Sent when any hard limit is tripped.
     bool: indicates if any limit is tripped
     Python list: the list shows all available joint's current limit values

Comment (silopolis) [reply docs/src/gui/gstat.adoc:193]: > But the compact notation has also advantages...

Agreed, but I believe structure brings more :-P

What about:

```
*`hard-limits-tripped`* - `(bool, list)`::
  * _Event_: Any hard limit is tripped. +
  * _Return_:
    * `bool`: boolean indicating if any limit is tripped,
    * `list`: Python list with all available joint's current limit values.
```

Comment (hansu) [reply docs/src/gui/gstat.adoc:193]: Looks good. But let's do this at a later stage. This should be basically aim at syncing, not improving the docs.

Comment (smoe) [reply docs/src/gui/gstat.adoc:193]: The sync may include a superior idea for the formatting that the translator came up with. And since now is the last opportunity to avoid also affect the upcoming translations, I kind of like to see such discussions and would eventually frame the style guide.

**Comment by silopolis** [comment]
Le jeu. 5 mai 2022 à 10:27, Steffen Möller ***@***.***> a
écrit :

> I kind of like to see such discussions and would eventually frame the
> style guide.
>
I'm pleased to discuss such things too. So much more than anchoring issues
! 😅
Speaking of style guide, I'm switching to it now, waiting for the sync to
go forward and pop up on Weblate :)

I've dumped my TODO and ideas in a GH project
https://github.com/users/silopolis/projects/2/views/3 to keep track of it
and see if it could be nice to use for collaboration. Taiga was nice, but
I'm not really willing to have another tool and platform.
But I'm having an issue not being able to link issues in my fork :/ Posted
messages in GH feedback and community forum...

File: docs/src/gui/gstat.adoc (hunk 5 above)
Comment (hansu) [reply docs/src/gui/gstat.adoc:193]: But this file has no counterpart in ES or FR, so it wouldn't affect translated files

**Comment by hansu** [comment]
I am done with the review. Jérémie you can cherry pick or merge my changes to your branch if you agree on them: https://github.com/hansu/linuxcnc/commit/92b66083c43c11333a6c8bed8a7cf4fb3e9ce6a9
(I can also commit them to your branch if that's ok for you)

**Comment by hansu** [comment]
>I've dumped my TODO and ideas in a GH project
https://github.com/users/silopolis/projects/2/views/3 to keep track of it
and see if it could be nice to use for collaboration.

Nice :slightly_smiling_face:

**Comment by hansu** [comment]
So I put all together in one commit on a new branch: https://github.com/hansu/linuxcnc/commit/77354ae3639232150f4fe27809b99bb554a66592
Hope that is okay now.

**Comment by silopolis** [comment]
@hansu merged and building ! Thanks for the review mate, that was a piece to swallow :sweat_smile: 

So, unless I lost myself, we're finally ready for The Merge ! :-)

**Comment by hansu** [comment]
You remember, we talked about cleaning up the docs-devel branch.
Maybe it is good to start by squashing your commits into some few. This reduces the clean-up task later.
I think it is not neccessary to have one commit for every folder.

**Comment by silopolis** [comment]
@hansu missed you in #linuxcnc-devel :'(

@andypugh do I have a GO to squash all individual folder sync commit into
one ? @hansu and @pere already agree on this...

@hansu I don't understand, do you want me to rebase en_es_fr_sync on master
? That's not how it should work, doesn't it ?
docs-devel should be rebased on master, then I do a final rebase on
docs-devel, you merge and create the PR for master... That's how I
understood it should be done.

**Comment by hansu** [comment]
Only squash your commits on your branch https://github.com/silopolis/linuxcnc/tree/docs/en_es_fr_sync a bit.

**Comment by hansu** [comment]
Thanks!

**Comment by andypugh** [comment]
On Fri, 6 May 2022 at 13:37, Jérémie Tarot ***@***.***> wrote:

>
> @andypugh do I have a GO to squash all individual folder sync commit into
> one ? @hansu and @pere already agree on this...


OK with me.

I would suggest keeping the full commit history for Submakefile though.

-- 
atp

**Comment by silopolis** [comment]
Le ven. 6 mai 2022 à 21:31, andypugh ***@***.***> a écrit :

> On Fri, 6 May 2022 at 13:37, Jérémie Tarot ***@***.***> wrote:
> > @andypugh do I have a GO to squash all individual folder sync commit into
> > one ? @hansu and @pere already agree on this...
>
> OK with me.
>
> I would suggest keeping the full commit history for Submakefile though.


Only squashed the adoc files sync commits so, unless mistaken, that should
be good

**Comment by silopolis** [comment]
Le jeu. 5 mai 2022 à 19:09, Hans ***@***.***> a écrit :

> I am done with the review. Jérémie you can cherry pick or merge my changes
> to your branch if you agree on them: ***@***.***
> <https://github.com/hansu/linuxcnc/commit/92b66083c43c11333a6c8bed8a7cf4fb3e9ce6a9>
> (I can also commit them to your branch if that's ok for you)
>
> If you can fix the couple of small  problems I've noted in comments, I'll
happily merge it :)

**Comment by silopolis** [comment]
Le ven. 6 mai 2022 à 09:45, Hans ***@***.***> a écrit :

> You remember, we talked about cleaning up the docs-devel branch.
> Maybe it is good to start by squashing your commits into some few. This
> reduces the clean-up task later.
> I think it is not neccessary to have one commit for every folder.
>
Well, that wasn't exactly @andypugh PoV. But I don't have enough git
background to judge.
Like I said I've already made a lot of reordering and squashing to get this
branch to this state, making the many passes and fixes look like a single
clean one.
I don't really care about squashing all the "folders commits" in one, but
may be it is best to do it in docs-devel ? Then we'd have my
en_es_fr_sync_dev with all the mess, my en_es_fr_sync cleaned up and well
organized and finally a concise docs-devel to be merged into master.
But I can make a backup of this branch at this stage before squashing...
You tell me.
